### DISCUSSION TITLE
Feature Request: custom input for arguments to -run when running 'Go: Test *' / "Go: Debug" command

### DISCUSSION THREAD

**Discussion by rucciva** [golang/vscode-go#1266] [Discussions]
For running specific subtests from a reusable test package (which consists of multi-level subtests). 

for example, to run only `TestSuite/Suite/Sub1/Sub1.1` in `impl1` package

```
package testsuite

type Interface interface{}

func Suite(t *testing.T,  i Interface) {
    t.Run("Sub1", func(t *testing.T){
        t.Run("Sub1.1", func(t *testing.T){
    
        })
        t.Run("Sub1.2", func(t *testing.T){
    
        })
    })
    t.Run("Sub2", func(t *testing.T){
        t.Run("Sub2.1", func(t *testing.T){
    
        })        
        t.Run("Sub2.2", func(t *testing.T){
    
        })
    })
}
```

```
package impl1


type impl struct  {
}

func TestSuite(t *testing.T) {
    t.Run("Suite",  func(t *testing.T) {
        teststuite.Suite(t, impl{})
    })
}
```

**Comment by hyangah** [comment]
@rucciva Do you mean that you want to have a text input shown when running `Go: Test Package` for example, and enter "-run=TestSuite/Suite/Sub1/Sub1.1"? Or do you want something more complicated than that?

If the former, I think one option is to use [custom tasks](https://code.visualstudio.com/docs/editor/tasks) that runs `go test` and takes [inputs](https://code.visualstudio.com/docs/editor/variables-reference#_input-variables).

**Reply by rucciva** [reply]
the former, where the default value for -run is displayed and editable. 

If using custom task, can we integrate with the debugging window of vscode or highlight tested code from the coverage report?
